src/hooks/useDeepChangeEffect.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, DependencyList } from 'react';
+
+import isEqual from 'lodash/isEqual';
+
+import { usePrevious } from 'hooks';
+
+/**
+ * It's like useEffect but runs when the deps list has a deep change.
+ * See: {@link isEqual}.
+ */
+export function useDeepChangeEffect(func: () => void, deps: DependencyList) {
+  const prev = usePrevious(deps);
+  useEffect(() => {
```

Comment:
should this support hook cleanup? https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

Reply:
might as well